src/sourceTracking.ts

```diff
@@ -605,7 +605,13 @@ export class SourceTracking extends AsyncCreatable {
     await Promise.all([
       this.updateLocalTracking({
         // assertion allowed because it's filtering out undefined
-        files: successes.map((fileResponse) => fileResponse.filePath as string).filter(Boolean),
+        files: successes
+          .filter((fileResponse) => fileResponse.state !== ComponentStatus.Deleted)
+          .map((fileResponse) => fileResponse.filePath as string)
+          .filter(Boolean),
+        deletedFiles: successes
```

Comment:
do we need to `.filter(Boolean)` on the `deletedFiles` too?

Reply:
@mshanemc - I don't know why `.filter(Boolean)` is necessary at all but maybe you do?  I wouldn't think the filter would be needed for the deletedFiles but maybe it does?

Reply:
this line hints at the issue...SDR's types have filePath as optional.  You can see that `as` trying to convine TS that there's a string there.

```ts
          .map((fileResponse) => fileResponse.filePath as string)
```

filter(Boolean) will take out any `undefined` from the array.